Marlin/src/gcode/config/M43.cpp

```diff
@@ -141,7 +141,7 @@ inline void servo_probe_test() {
 
     SERIAL_PROTOCOLLNPGM(". deploy & stow 4 times");
     SET_INPUT_PULLUP(PROBE_TEST_PIN);
-    bool deploy_state, stow_state;
+    bool deploy_state = false, stow_state = false;
```

Comment:
Bytes are always at a premium, so these initializers can be dropped if there's no pressing reason to have them. They are set in the following loop, and we only care about the last values read.

Reply:
I added the initializers to eliminate a "may be used uninitialized" warning. You and I can see that it's obvious that they are written to before ever being written from, but apparently there's an issue with GCC where it assumes the loop may never run (even though it has a fixed length and will always run 4 times), so it generates the warning. Any suggestions on other ways to clear that up?

Reply:
Here's some further detail on why the warning is generated:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=36296

Reply:
The compiler complains because the second clause of the `for` loop comes ahead of the loop. Using a `do { ... } while()` loop instead will prevent the warning.

Reply:
Let me know if the change I made is what you had in mind.